zebra-consensus/src/checkpoint/list.rs

```diff
@@ -53,6 +57,13 @@ impl CheckpointList {
             _ => Err("checkpoints must start at the genesis block height 0")?,
         };
 
+        // Check the hash of the genesis block against the supplied network
+        match checkpoints.values().next() {
+            Some(h) if h == &parameters::genesis_hash(network) => {}
+            Some(_) => Err("the genesis checkpoint does not match the network genesis hash")?,
+            _ => unreachable!("already checked for an empty list"),
```

Comment:
Rather than having an `unreachable!` I think this could be combined with the preceding check, as
```
match checkpoints.iter().next() {
    Some((BlockHeight(0), h)) if h == genesis_hash(network) => {}
    Some((BlockHeight(0), _)) => Err("the genesis checkpoint does not match the network genesis hash")?,
    Some((BlockHeight(_), _)) => Err("checkpoints must start at the genesis block height 0")?,
    None => Err("checkpoint lists must have at least one checkpoint")?,
}
```
to get one exhaustive check.

Reply:
Fixed in 66f2bf64702921f2b845f3bcaa9f7ae892f7fc5e.